run.py

```diff
@@ -240,140 +259,114 @@ def main():
+    nonbib_db_conn = nonbib_db_engine.connect()
+
+    metrics_connection_string = config.get('METRICS_DATABASE',
+                                   'postgresql://postgres@localhost:5432/postgres')
+    metrics_db_engine = create_engine(metrics_connection_string)
+    metrics_db_conn = metrics_db_engine.connect()
+
+    if args.command == 'help' and args.diagnose:
+        diagnose_nonbib()
+        diagnose_metrics()
+
+    elif args.command == 'createIngestTables':
+        sql_sync = nonbib.NonBib(args.rowViewSchemaName)
+        sql_sync.create_column_tables(nonbib_db_engine)
 
-    if args.command == 'createIngestTables':
-        sql_sync = row_view.SqlSync(args.rowViewSchemaName, config)
-        sql_sync.create_column_tables()
     elif args.command == 'dropIngestTables':
-        sql_sync = row_view.SqlSync(args.rowViewSchemaName, config)
-        sql_sync.drop_column_tables()
+        sql_sync = nonbib.NonBib(args.rowViewSchemaName)
+        sql_sync.drop_column_tables(nonbib_db_engine)
+
     elif args.command == 'createJoinedRows':
-        sql_sync = row_view.SqlSync(args.rowViewSchemaName, config)
-        sql_sync.create_joined_rows()
-
-    elif args.command == 'ingestMeta' and args.rowViewSchemaName:
-        # run copy from program command
-        # that sql command will invoke ingest of each table
-        sql_sync = row_view.SqlSync(args.rowViewSchemaName, config)
-        Session = sessionmaker()
-        sess = Session(bind=sql_sync.connection)
-        for t in row_view.SqlSync.all_types:
-            command_args = ' ingest --fileType ' + t +   \
-                ' --rowViewSchemaName ' + args.rowViewSchemaName
-            python_command = "'python " + os.path.abspath(__file__) + command_args + "'"
-            sql_command = 'copy ' + args.rowViewSchemaName + '.' + t +  \
-                ' from program ' + python_command + ';'
-
-            sess.execute(sql_command)
-            sess.commit()
-
-        sess.close()
+        sql_sync = nonbib.NonBib(args.rowViewSchemaName, config)
+        sql_sync.create_joined_rows(nonbib_db_conn)
 
     elif args.command == 'createMetricsTable' and args.metricsSchemaName:
         m = metrics.Metrics(args.metricsSchemaName)
-        m.create_metrics_table()
+        m.create_metrics_table(metrics_db_engine)
+
     elif args.command == 'dropMetricsTable' and args.metricsSchemaName:
         m = metrics.Metrics(args.metricsSchemaName)
-        m.drop_metrics_table()
+        m.drop_metrics_table(metrics_db_engine)
+
     elif args.command == 'populateMetricsTable' and args.rowViewSchemaName and args.metricsSchemaName:
-        # m = metrics.Metrics(None, {'FROM_SCRATCH': False, 'COPY_FROM_PROGRAM': False})
-        m = metrics.Metrics(None, {'FROM_SCRATCH': True, 'COPY_FROM_PROGRAM': False})
-        m.update_metrics_all(args.rowViewSchemaName)
+        m = metrics.Metrics()
+        m.update_metrics_all(metrics_db_conn, nonbib_db_conn, args.rowViewSchemaName)
```

Comment:
this call also loads all results into memory before copying them; please change all occurences where a large datasets are being loaded into memory

also, I'm seeing the following being repeated on many occassions

```
Session = sessionmaker()
sess = Session(bind=db_conn)
```

you are passing the db connections to methods of a metrics/nonbib instance; consider initializing these instances with the db connections (i.e. let them establish connection and keep the session) - that way perhaps many lines can be dropped fromthe code

Reply:
all results are no longer loaded into memory: https://github.com/spacemansteve/AdsDataSqlSync/blob/master/adsdata/metrics.py#L177

Reply:
addressed in new commit: https://github.com/spacemansteve/AdsDataSqlSync/commit/04a256b429c48c444a8984d69487f50ee6aff8b2

Reply:
regarding db connections, I decided to have the calling function pass in the connection so it is clear the calling function was also responsible for closing the connection.  I will investigate best practices and existing project code to decide if/how to change.